src/bin/scheduler/setup/directories.rs

```diff
@@ -368,6 +370,66 @@ fn clean_up_results_directory(
     Ok(results_directory_lock.release()?)
 }
 
+struct OwnershipSetter {
+    #[cfg(unix)]
+    user_id: u32,
+    #[cfg(unix)]
+    group_id: u32,
+    #[cfg(windows)]
+    user: String,
+}
+
+impl OwnershipSetter {
+    pub fn make_for_current_user() -> AnyhowResult<Self> {
+        #[cfg(unix)]
+        {
+            Ok(Self {
+                user_id: unsafe { libc::getuid() },
+                group_id: unsafe { libc::getgid() },
+            })
+        }
+        #[cfg(windows)]
+        {
+            Ok(Self {
+                user: whoami::username(),
+            })
+        }
+    }
+
+    pub fn take_ownership(&self, target: &Utf8Path, recursive: bool) -> AnyhowResult<()> {
+        #[cfg(unix)]
+        {
+            (if recursive {
+                self.take_ownership_recursive_unix(target)
+            } else {
+                std::os::unix::fs::chown(target, Some(self.user_id), Some(self.group_id))
+            })
+            .context(format!(
+                "Failed to set ownership of {target} to `{}:{}`",
+                self.user_id, self.group_id
+            ))
+        }
+        #[cfg(windows)]
+        {
+            use super::windows_permissions::set_ownership;
+            set_ownership(&self.user, target, recursive)
+        }
+    }
+
+    #[cfg(unix)]
+    fn take_ownership_recursive_unix(&self, target: &Utf8Path) -> std::io::Result<()> {
```

Comment:
Dont include `unix` in the function name. This information will always be clear to the reader anyway.